Use minsdk 24 and update docs (#149)
elastic/apm-agent-android: docs/setup.asciidoc

NOTE: When both `secretToken` and `apiKey` are provided, apiKey has priority and secretToken is ignored.

[float]
[[minsdk-24-support]]
==== For projects using minSdk version < 26

> Due to Android's limited support for Java 8 features on devices with API level < 26, or in other words, older than Android 8.0, it is required to add https://developer.android.com/studio/write/java8-support#library-desugaring[Java 8+ desugaring support] to apps which minSdk version is lower than 26.
> Otherwise, your app could crash when running on devices using Android OS versions older than 8.0. This is due to the fact that the https://github.com/open-telemetry/opentelemetry-java[OpenTelemetry Java SDK], which this SDK is built upon, makes use of some of those missing Java 8 features.
> 
> If your minSdk is 26 or higher though, you don't need to worry about adding the Java 8+ desugaring support.

Reviewer: Wording suggestion

```suggestion
Due to Android's limited support for Java 8 features on devices with API level < 26, or in other words, older than Android 8.0, you must add https://developer.android.com/studio/write/java8-support#library-desugaring[Java 8+ desugaring support] to apps with a minSdk version lower than 26.
If you don't, your app can crash when running on devices using Android OS versions older than 8.0. This is because the https://github.com/open-telemetry/opentelemetry-java[OpenTelemetry Java SDK], which this SDK is built upon, uses Java 8 features.

If your minSdk is 26 or higher, you can skip this step.
```
